### PR TITLE
[minor][runners] support inline custom rules

### PR DESCRIPTION
**Pull request by ealush**
## Inline Custom Rules
There is another way to add rules per a single run that does not involve the hassle of creating a new rules object. Instead, the function can be written in-line, along with the other selected rules. Just as in all other rules, the value supplied to the function is the value being enforced.

```js
    Passable('GroupName', (pass, enforce) => {
        pass('TestName', 'Must have a valid email', () => {
            enforce(user.email).allOf({
                isValidEmail: (value) => value.indexOf('@') > -1
            });
        });
    });
```

**Note**: If a rule with the selection function name already exists, it will take precedence over the inline function, in which case, your tests may fail as you are supplying function value where it is not expected.

### CODE REVIEW

File: src/enforce/runners/run/index.js [hunk 1]
```diff
@@ -1,7 +1,22 @@
 // @flow
+/**
+ * A function which returns whether a combination of
+ * rule + value is true or false
+ *
+ * @param {any} value - the value being tested
+ * @param {string} key the name of the rule being run
+ * @param {string} tests an object containing the group of tests in the current run
```
Comment (omrilotan) [inline src/enforce/runners/run/index.js]: tests and rules aren't strings.
To define types in jsdoc, see [typedef](http://usejsdoc.org/tags-typedef.html)

File: src/enforce/runners/run/index.js [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
+ * @param {string} rules an object containing all the rules available for the current pass
+ * @return {boolean} value validation result
+ */
```
Comment (raphaelboukara) [inline src/enforce/runners/run/index.js]: which plugin are you using to generate those kind of description before your functions?

Comment (raphaelboukara) [reply src/enforce/runners/run/index.js]: @ealush

File: package.json [hunk 3]
```diff
@@ -31,20 +31,20 @@
...
     "cross-env": "^5.1.1",
-    "eslint": "^4.5.0",
+    "eslint": "^4.11.0",
     "eslint-loader": "^1.9.0",
     "eslint-plugin-flowtype": "^2.35.0",
-    "flow-bin": "^0.57.3",
+    "flow-bin": "^0.59.0",
     "mocha": "^4.0.1",
-    "webpack": "^3.5.5"
+    "webpack": "^3.8.1"
```
Comment (raphaelboukara) [inline package.json]: seems all those updates are not linked to your commit, why not create a specific commit update?

File: src/enforce/runners/run/index.js (hunk 1 above)
Comment (ealush) [reply src/enforce/runners/run/index.js]: Whoops. Duplicated the line and forgot to edit the types. Good catch

File: src/enforce/runners/run/index.js (hunk 2 above)
Comment (ealush) [reply src/enforce/runners/run/index.js]: It's called [JSDoc](http://usejsdoc.org/). Supported by github and most code editors.

**Pull request merged**